### PR TITLE
Improving the Jakarta NoSQL JavaDoc

### PR DESCRIPTION
**Pull request by dearrudam**
As I've mentioned at https://github.com/jakartaee/nosql/pull/166#issuecomment-1585689112, I've reviewed the API's JavaDoc.

## Changes:
- Enhance the `@jakarta.nosql.Entity` javadoc;
- Enhance the `@jakarta.nosql.Column` javadoc;
- Enhance the `@jakarta.nosql.Id` javadoc;
- Enhance the `jakarta.nosql.Template` javadoc;
- Enhance the `jakarta.nosql.QueryMapper` javadoc;
- Enhance the `jakarta.nosql.PreparedStatement` javadoc;
- Enhance the `jakarta.nosql.column` package javadoc;
- Enhance the `jakarta.nosql.column.ColumnTemplate` javadoc;
- Enhance the `jakarta.nosql.keyvalue` package javadoc;
- Enhance the `jakarta.nosql.keyvalue.KeyValueTemplate` javadoc;
- Enhance the `jakarta.nosql.document` package javadoc;
- Enhance the `jakarta.nosql.document.DocumentTemplate` javadoc;

### CODE REVIEW

**Comment by dearrudam** [comment]
@otaviojava, please, could you review this PR!?

File: api/nosql-core/src/main/java/jakarta/nosql/Entity.java [hunk 1]
```diff
@@ -22,13 +22,76 @@
 import java.lang.annotation.Target;
 
 /**
- * Specifies that the class is an entity. This annotation is applied to the entity class.
+ * Specifies that the class is an entity.
+ * <p>
+ * You can include one or multiple entities without requiring additional annotations like {@code OneToOne} or {@code OneToMany} in JPA when using the API.
```
Comment (otaviojava) [inline api/nosql-core/src/main/java/jakarta/nosql/Entity.java:27]:
```suggestion
´´´
I recommend not putting any information on JPA, NoSQL is different, and I am afraid to put confusion or misunderstanding.

File: api/nosql-column/src/main/java/jakarta/nosql/column/ColumnTemplate.java [hunk 2]
```diff
@@ -56,6 +61,9 @@ public interface ColumnTemplate extends Template {
     /**
      * Executes a native query database then bring the result as a {@link Stream}
      *
+     * <p>
```
Comment (otaviojava) [inline api/nosql-column/src/main/java/jakarta/nosql/column/ColumnTemplate.java:64]: Max where are you closing this tag?

File: api/nosql-column/src/main/java/jakarta/nosql/column/ColumnTemplate.java [hunk 3]
```diff
@@ -67,6 +75,9 @@ public interface ColumnTemplate extends Template {
     /**
      * Executes a query then bring the result as a unique result
      *
+     * <p>
```
Comment (otaviojava) [inline api/nosql-column/src/main/java/jakarta/nosql/column/ColumnTemplate.java:78]: The same here

File: api/nosql-column/src/main/java/jakarta/nosql/column/ColumnTemplate.java (hunk 2 above)
Comment (dearrudam) [reply api/nosql-column/src/main/java/jakarta/nosql/column/ColumnTemplate.java:65]: I've just fixed it!

File: api/nosql-column/src/main/java/jakarta/nosql/column/ColumnTemplate.java (hunk 3 above)
Comment (dearrudam) [reply api/nosql-column/src/main/java/jakarta/nosql/column/ColumnTemplate.java:80]: I've just fixed it!

**Review by otaviojava — APPROVED** [review]

**Pull request merged**